api/tests/test_auth.rs

```diff
@@ -3,17 +3,39 @@ mod tests {
     use actix_web::http::StatusCode;
     use actix_web::middleware::from_fn;
     use actix_web::{App, test, web};
-    use boom_api::auth::{auth_middleware, get_default_auth};
-    use boom_api::conf::AppConfig;
-    use boom_api::db::get_default_db;
+    use boom_api::auth::{auth_middleware, get_test_auth};
+    use boom_api::conf::{AppConfig, load_dotenv};
+    use boom_api::db::get_test_db;
     use boom_api::routes;
     use mongodb::{Database, bson::doc};
 
     /// Test POST /auth
     #[actix_rt::test]
     async fn test_post_auth() {
-        let database: Database = get_default_db().await;
-        let auth_app_data = get_default_auth(&database).await.unwrap();
+        load_dotenv();
+
+        // Clear users collection for clean test BEFORE setting up the database
+        let config = AppConfig::from_path("../tests/config.test.yaml");
+        let db_config = config.database;
+        let uri = std::env::var("MONGODB_URI").unwrap_or_else(|_| {
+            format!(
+                "mongodb://{}:{}@{}:{}",
+                db_config.username, db_config.password, db_config.host, db_config.port
+            )
+        });
+        let client = mongodb::Client::with_uri_str(uri)
+            .await
+            .expect("failed to connect");
+        let temp_db = client.database(&db_config.name);
+        temp_db
+            .collection::<mongodb::bson::Document>("users")
+            .drop()
+            .await
+            .ok(); // Ignore errors if collection doesn't exist
+
```

Comment:
This is a problem. Our tests are async i.e. they can run in parallel and wiping a table clean (which is used by multiple tests) can add flakiness and create unexpected results.

Reply:
+ that lengthy-ish piece of code is duplicated btw